ios/Capacitor/Capacitor/CapacitorBridge.swift

```diff
@@ -537,6 +537,60 @@ internal class CapacitorBridge: NSObject, CAPBridgeProtocol {
         }
     }
 
+    // MARK: - CAPBridgeProtocol: Paths, Files, Assets
+
+    /**
+     * Translate a URL from the web view into a file URL for native iOS.
+     *
+     * The web view may be handling several different types of URLs:
+     *   - res:// (shortcut scheme to web assets)
+     *   - file:// (fully qualified URL to file on the local device)
+     *   - base64:// (to be implemented)
+     *   - [web view scheme]:// (already converted once to load in the web view, to be implemented)
+     */
+    public func localURL(fromWebURL webURL: URL?) -> URL? {
+        guard let inputURL = webURL else {
+            return nil
+        }
+
+        let manager = FileManager.default
+        let url: URL
+
+        switch inputURL.scheme {
+        case "res":
+            let bundle = Bundle.main
+            let resourcePath = bundle.resourcePath!
+            url = URL(fileURLWithPath: resourcePath + "/public" + inputURL.path)
```

Comment:
URL's path manipulation methods are more robust than string modifications. Also, force-unwraps are bad.
```suggestion
            guard let resourcePath = Bundle.main.resourcePath else {
                return nil
            }
            url = URL(fileURLWithPath: resourcePath).appendingPathComponent("public").appendingPathComponent(inputURL.path)
```